### PR TITLE
[JBPM-9410] No Model definition in KIE Server Swagger API

### PR DESCRIPTION
**Pull request by fjtirado**
Moving response to ApiResponse

**JIRA**: 

[JBPM-9410](https://issues.redhat.com/browse/JBPM-9410)

### CODE REVIEW

**Review by elguardian — CHANGES_REQUESTED** [review]
please use jbpm jira, not rhpam

**Comment by fjtirado** [comment]
> please use jbpm jira, not rhpam

Can you specify a clear criteria for using RHPAM or JBPM (since for other RHPAM we have used directly the RHPAM to commit changes? Once a PR is created, switching from RHPAM to JBPM is a burden.

**Review by elguardian — APPROVED** [review]
Ok. LGTM please try to avoid reformatting imports when they are not needed.

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/RuntimeDataResource.java [hunk 1]
```diff
@@ -768,7 +742,7 @@ public Response getTasksByVariables(@Context HttpHeaders headers,
         }
     }
 
-    @ApiOperation(value = "Queries processes by variables and tasks")
+    @ApiOperation(value = "Queries processes by variables and tasks", response = ProcessInstanceCustomVarsList.class)
```
Comment (afalhambra) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/RuntimeDataResource.java:745]: For consistency, maybe we can use this PR to include a 200 response code for this method.

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/RuntimeDataResource.java [hunk 2]
```diff
@@ -794,7 +768,7 @@ public Response queryProcessesByVariables(@Context HttpHeaders headers, String p
 
     }
 
-    @ApiOperation(value = "Queries process tasks by variables")
+    @ApiOperation(value = "Queries process tasks by variables", response = ProcessInstanceUserTaskWithVariablesList.class)
```
Comment (afalhambra) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/RuntimeDataResource.java:771]: same as above

**Review by afalhambra — CHANGES_REQUESTED** [review]
Very nice work. The only remarkable change I'm missing is the `kie-server-rest-case-mgmt` module which has not been modified to include this change.

I also wanted to know what @MarianMacik thinks regarding revapi config module, though I think that should be ok.

Thanks @fjtirado!

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/pom.xml [hunk 3]
```diff
@@ -174,6 +174,9 @@
       <plugin>
         <groupId>org.revapi</groupId>
         <artifactId>revapi-maven-plugin</artifactId>
+        <configuration>
+          <failSeverity>breaking</failSeverity>
+        </configuration>
```
[the remark this answers is not in the document]
Comment (fjtirado) [reply kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/pom.xml:179]: ok, let me know if there is other way. Adding all ignores does not makes sense

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/DefinitionResource.java [hunk 4]
```diff
@@ -93,11 +92,10 @@ public DefinitionResource(DefinitionServiceBase definitionServiceBase, KieServer
     }
 
 
-    @ApiOperation(value="Returns entity and task information for a specified process.",
-            response=ProcessDefinition.class, code=200)
+    @ApiOperation(value="Returns entity and task information for a specified process.")
     @ApiResponses(value = { @ApiResponse(code = 500, message = "Unexpected error"),
             @ApiResponse(code = 404, message = "Process or Container Id not found"), 
-            @ApiResponse(code = 200, message = "Successfull response", examples=@Example(value= {
+            @ApiResponse(code = 200, response = ProcessDefinition.class, message = "Successfull response", examples=@Example(value= {
```
[the remark this answers is not in the document]
Comment (fjtirado) [reply kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/DefinitionResource.java:98]: sure, changing them all

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/RuntimeDataResource.java (hunk 1 above)
Comment (fjtirado) [reply kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/RuntimeDataResource.java:745]: 200 is the default, so it is implicit

**Comment by fjtirado** [comment]
> Very nice work. The only remarkable change I'm missing is the `kie-server-rest-case-mgmt` module which has not been modified to include this change.
> 
Sure, Ill change those ones too

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-case-mgmt/src/main/java/org/kie/server/remote/rest/casemgmt/CaseAdminResource.java [hunk 5]
```diff
@@ -79,10 +79,9 @@ public CaseAdminResource(
         this.caseAdminServiceBase = caseAdminServiceBase;
     }
 
-    @ApiOperation(value="Returns case instances without authentication checks.",
-            response=CaseInstanceList.class, code=200)
+    @ApiOperation(value="Returns case instances without authentication checks.")
     @ApiResponses(value = { @ApiResponse(code = 500, message = "Unexpected error"), 
-            @ApiResponse(code = 200, message = "Successfull response", examples=@Example(value= {
+            @ApiResponse(code = 200, response = CaseInstanceList.class, message = "Successfull response", examples=@Example(value= {
```
Comment (afalhambra) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-case-mgmt/src/main/java/org/kie/server/remote/rest/casemgmt/CaseAdminResource.java:84]: Sorry about @fjtirado - seems you changed this successful message for all modules except for this case-mgmt one. Do you mind updating this as well?
Thanks again

**Review by afalhambra — APPROVED** [review]
Great work. Only successful message hasn't been updated for case-mgmt. Sorry about it.
Other than that it's perfect.

**Review by afalhambra — COMMENTED** [review]
Great work! thanks a lot @fjtirado

**Review by afalhambra — APPROVED** [review]

**Review by gmunozfe — APPROVED** [review]
looks good, fantastic work @fjtirado !

**Comment by fjtirado** [comment]
@kiegroup/gatekeepers please merge this

**Pull request merged**